guacamole-common-js/src/main/webapp/modules/Client.js

```diff
@@ -1699,11 +1692,63 @@ Guacamole.Client = function(tunnel) {
             tunnel.sendMessage("nop");
         }, 5000);
 
-        setState(STATE_WAITING);
+        setState(Guacamole.Client.State.WAITING);
     };
 
 };
 
+/**
+ * All possible Guacamole Client states.
+ * 
+ * @type {!Object.<string, number>}
+ */
+Guacamole.Client.State = {
+    
+    /**
+     * The client is idle, with no active connection.
+     * 
+     * @type number
```

Comment:
I believe this (and the others) needs to be: `@type {!number}`

See: https://jsdoc.app/tags-type.html

Reply:
Okay, should be fixed.